tests/test_study.py

```diff
@@ -1010,3 +1012,58 @@ def test_study_summary_lt_le():
     summaries.sort()
     assert summaries[0] == summary_0
     assert summaries[1] == summary_1
+
+
+def test_log_completed_trial_verbosity(capsys: _pytest.capture.CaptureFixture) -> None:
```

Comment:
Thanks for adding those tests! They doesn't exactly verify the introduced changes to the logic but I think they are good to keep.

Reply:
Thank you for your comment. I also added the logic which skips storage access in https://github.com/optuna/optuna/pull/1403/commits/22ccbff53f36292507c2084edad4f26222ab32b0. I noticed that `_log_completed_trial` accessed twice to get the best trial. I think we can reduce it by reusing the first one, but it may be out of the scope of this PR.